src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateNew = JSON.parse(JSON.stringify(state));
```

Comment:
[CODE STYLE] - Avoid using `JSON.parse(JSON.stringify(obj))` to clone objects because it can lead to performance issues and it doesn't copy methods or handle circular references. Use the spread operator or `Object.assign` instead.